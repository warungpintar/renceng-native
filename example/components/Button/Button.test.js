import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {Button, Variants, Text, Icon} from '@warungpintar/renceng-native';

const {ButtonVariants, ButtonSizes} = Variants;

describe('Button', () => {
  describe('use variant props', () => {
    const mockFn = jest.fn(() => null);
    const primaryProps = {
      testID: 'myPrimaryButton',
      variant: 'primary',
      size: 'md',
      onPress: mockFn,
    };
    /*
      FIXME: This test uses render at every case since
      it's unmounted before it finish running each cases
    */
    it('outputs correct title/text', () => {
      const {getByText, toJSON} = render(
        <Button {...primaryProps}>
          <Text>My Button</Text>
        </Button>,
      );

      expect(getByText('My Button')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    });

    it('fires the assigned event on press', async () => {
      const {queryByTestId, toJSON} = render(<Button {...primaryProps} />);
      const primaryButton = queryByTestId('myPrimaryButton');
      fireEvent.press(primaryButton);
      fireEvent.press(primaryButton);

      await waitFor(() => expect(mockFn.mock.calls.length).toBe(2));
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders primary variant correctly', () => {
      const {queryByTestId, toJSON} = render(<Button {...primaryProps} />);
      const btnProps = queryByTestId('myPrimaryButton').props;

      expect(btnProps.style?.[0]?.backgroundColor).toMatch(
        ButtonVariants.variants.primary.bg,
      );
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders successfully', () => {
      const activeButton = render(<Button {...primaryProps} />);
      const {toJSON} = activeButton;

      expect(activeButton).toBeDefined();
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('use size props correctly', () => {
    const mockFn = jest.fn(() => null);
    const primaryProps = {
      testID: 'myPrimaryButton',
      variant: 'primary',
      size: 'lg',
      onPress: mockFn,
    };

    it('outputs correct title/text', () => {
      const {getByText, toJSON} = render(
        <Button {...primaryProps}>
          <Text>My Button</Text>
        </Button>,
      );

      expect(getByText('My Button')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    });

    it('fires the assigned event on press', async () => {
      const {queryByTestId, toJSON} = render(<Button {...primaryProps} />);
      const primaryButton = queryByTestId('myPrimaryButton');
      fireEvent.press(primaryButton);
      fireEvent.press(primaryButton);

      await waitFor(() => expect(mockFn.mock.calls.length).toBe(2));
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders primary variant and large size correctly', () => {
      const {queryByTestId, toJSON} = render(<Button {...primaryProps} />);
      const btnProps = queryByTestId('myPrimaryButton').props;

      expect(btnProps.style?.[0]?.backgroundColor).toMatch(
        ButtonVariants.variants.primary.bg,
      );
      expect(btnProps.style?.[0]?.paddingTop).toBe(12);
      expect(btnProps.style?.[0]?.paddingBottom).toBe(12);
      expect(btnProps.style?.[0]?.paddingLeft).toBe(24);
      expect(btnProps.style?.[0]?.paddingRight).toBe(24);
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders successfully', () => {
      const activeButton = render(<Button {...primaryProps} />);
      const {toJSON} = activeButton;

      expect(activeButton).toBeDefined();
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('in disabled/inactive state', () => {
    const mockFn = jest.fn(() => null);
    const disabledProps = {
      testID: 'myDisabledButton',
      variant: 'primary',
      size: 'md',
      onPress: mockFn,
      disabled: true,
    };

    it('outputs correct title/text', () => {
      const {getByText, toJSON} = render(
        <Button {...disabledProps}>
          <Text>My Button</Text>
        </Button>,
      );

      expect(getByText('My Button')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    });

    it("doesn't fire event on press", async () => {
      const {queryByTestId, toJSON} = render(<Button {...disabledProps} />);
      const myDisabledButton = queryByTestId('myDisabledButton');
      fireEvent.press(myDisabledButton);

      await waitFor(() => expect(mockFn.mock.calls.length).toBe(0));
      expect(toJSON()).toMatchSnapshot();
    });

    it('shows correct color when disabled', () => {
      const {queryByTestId, toJSON} = render(<Button {...disabledProps} />);
      const btnProps = queryByTestId('myDisabledButton').props;

      expect(btnProps.style?.[0]?.backgroundColor).toMatch(
        ButtonVariants.variants.disabled.bg,
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('with icon', () => {
    const mockFn = jest.fn(() => null);
    const buttonProps = {
      testID: 'myIconButton',
      variant: 'primary',
      size: 'md',
      onPress: mockFn,
    };

    const iconProps = {
      testID: 'my-icon',
      name: 'car-outline',
      size: 16,
    };

    it('renders icon component', () => {
      const {queryAllByTestId, toJSON} = render(
        <Button {...buttonProps}>
          <Icon {...iconProps} />
        </Button>,
      );
      const iconElement = queryAllByTestId('my-icon');

      expect(iconElement.length).toBe(1);
      expect(toJSON()).toMatchSnapshot();
    });

    it("shouldn't render icon component if icon property is empty", () => {
      const noIconProps = {
        testID: 'noIconButton',
        variant: 'primary',
        size: 'md',
        title: 'My Button',
        onPress: mockFn,
      };
      const {queryAllByTestId, toJSON} = render(<Button {...noIconProps} />);
      const iconElement = queryAllByTestId('my-icon');

      expect(iconElement.length).toBe(0);
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders successfully', () => {
      const activeButton = render(<Button {...iconProps} />);
      const {toJSON} = activeButton;

      expect(activeButton).toBeDefined();
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('in loading state', () => {
    const mockFn = jest.fn(() => null);
    const loadingProps = {
      testID: 'myLoadingButton',
      variant: 'primary',
      size: 'md',
      title: 'My Button',
      onPress: mockFn,
      isLoading: true,
    };

    it('should render loading spinner', () => {
      const {toJSON, getByA11yHint} = render(<Button {...loadingProps} />);
      const activityIndicator = getByA11yHint('loading');

      expect(activityIndicator).toBeDefined();
      expect(toJSON()).toMatchSnapshot();
    });

    it('should be in disabled state and not triggering any action', async () => {
      const {toJSON, getByTestId} = render(<Button {...loadingProps} />);
      const btn = getByTestId('myLoadingButton');
      fireEvent.press(btn);

      expect(btn.props.isLoading).toBeTruthy();
      await waitFor(() => expect(mockFn.mock.calls.length).toBe(0));
      expect(toJSON()).toMatchSnapshot();
    });

    it('renders successfully', () => {
      const activeButton = render(<Button {...loadingProps} />);
      const {toJSON} = activeButton;

      expect(activeButton).toBeDefined();
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
