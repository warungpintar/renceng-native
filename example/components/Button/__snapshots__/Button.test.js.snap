// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button in disabled/inactive state doesn't fire event on press 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E8E8E8",
        "borderColor": "#E8E8E8",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#737373",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myDisabledButton"
  variant="disabled"
/>
`;

exports[`Button in disabled/inactive state outputs correct title/text 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E8E8E8",
        "borderColor": "#E8E8E8",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#737373",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myDisabledButton"
  variant="disabled"
>
  <Text
    color="#161616"
    fontFamily="OpenSans-Regular"
    style={
      Array [
        Object {
          "color": "#161616",
          "fontFamily": "OpenSans-Regular",
        },
        undefined,
      ]
    }
  >
    My Button
  </Text>
</View>
`;

exports[`Button in disabled/inactive state shows correct color when disabled 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#E8E8E8",
        "borderColor": "#E8E8E8",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#737373",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myDisabledButton"
  variant="disabled"
/>
`;

exports[`Button in loading state renders successfully 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myLoadingButton"
  title="My Button"
  variant="primary"
>
  <View
    alignItems="center"
    flexDirection="row"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <ActivityIndicator
      accessibilityHint="loading"
      color="#161616"
      size="small"
    />
  </View>
</View>
`;

exports[`Button in loading state should be in disabled state and not triggering any action 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myLoadingButton"
  title="My Button"
  variant="primary"
>
  <View
    alignItems="center"
    flexDirection="row"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <ActivityIndicator
      accessibilityHint="loading"
      color="#161616"
      size="small"
    />
  </View>
</View>
`;

exports[`Button in loading state should render loading spinner 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myLoadingButton"
  title="My Button"
  variant="primary"
>
  <View
    alignItems="center"
    flexDirection="row"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <ActivityIndicator
      accessibilityHint="loading"
      color="#161616"
      size="small"
    />
  </View>
</View>
`;

exports[`Button use size props correctly fires the assigned event on press 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="lg"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 12,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 12,
      },
      undefined,
    ]
  }
  testID="myPrimaryButton"
  variant="primary"
/>
`;

exports[`Button use size props correctly outputs correct title/text 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="lg"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 12,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 12,
      },
      undefined,
    ]
  }
  testID="myPrimaryButton"
  variant="primary"
>
  <Text
    color="#161616"
    fontFamily="OpenSans-Regular"
    style={
      Array [
        Object {
          "color": "#161616",
          "fontFamily": "OpenSans-Regular",
        },
        undefined,
      ]
    }
  >
    My Button
  </Text>
</View>
`;

exports[`Button use size props correctly renders primary variant and large size correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="lg"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 12,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 12,
      },
      undefined,
    ]
  }
  testID="myPrimaryButton"
  variant="primary"
/>
`;

exports[`Button use size props correctly renders successfully 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="lg"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 12,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 12,
      },
      undefined,
    ]
  }
  testID="myPrimaryButton"
  variant="primary"
/>
`;

exports[`Button use variant props fires the assigned event on press 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myPrimaryButton"
  variant="primary"
/>
`;

exports[`Button use variant props outputs correct title/text 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myPrimaryButton"
  variant="primary"
>
  <Text
    color="#161616"
    fontFamily="OpenSans-Regular"
    style={
      Array [
        Object {
          "color": "#161616",
          "fontFamily": "OpenSans-Regular",
        },
        undefined,
      ]
    }
  >
    My Button
  </Text>
</View>
`;

exports[`Button use variant props renders primary variant correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myPrimaryButton"
  variant="primary"
/>
`;

exports[`Button use variant props renders successfully 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myPrimaryButton"
  variant="primary"
/>
`;

exports[`Button with icon renders icon component 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="myIconButton"
  variant="primary"
>
  <View
    collapsable={false}
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={16}
      bbWidth={16}
      color={4278190080}
      fill="black"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      size={16}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      testID="my-icon"
      tintColor={4278190080}
      vbHeight={24}
      vbWidth={24}
      width={16}
    >
      <RNSVGGroup
        fill={4278190080}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGGroup>
          <RNSVGGroup>
            <RNSVGRect
              height="24"
              opacity={0}
              width="24"
              x={0}
              y={0}
            />
            <RNSVGPath
              d="M21.6 11.22L17 7.52V5a1.91 1.91 0 0 0-1.81-2H3.79A1.91 1.91 0 0 0 2 5v10a2 2 0 0 0 1.2 1.88 3 3 0 1 0 5.6.12h6.36a3 3 0 1 0 5.64 0h.2a1 1 0 0 0 1-1v-4a1 1 0 0 0-.4-.78zM20 12.48V15h-3v-4.92zM7 18a1 1 0 1 1-1-1 1 1 0 0 1 1 1zm5-3H4V5h11v10zm7 3a1 1 0 1 1-1-1 1 1 0 0 1 1 1z"
            />
          </RNSVGGroup>
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`Button with icon renders successfully 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  name="car-outline"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size={16}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
      },
      undefined,
    ]
  }
  testID="my-icon"
  variant="primary"
/>
`;

exports[`Button with icon shouldn't render icon component if icon property is empty 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  alignItems="center"
  collapsable={false}
  focusable={true}
  isLoading={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  size="md"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FECF28",
        "borderColor": "#FECF28",
        "borderRadius": 4,
        "borderWidth": 1,
        "color": "#161616",
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  testID="noIconButton"
  title="My Button"
  variant="primary"
/>
`;
